File: leetcode-04/solutions/python3/0560.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans = 0
        sum = 0
        dict = collections.defaultdict(int)
        dict[0] = 1

        for num in nums:
            sum += num
            if sum - k in dict:
                ans += dict[sum - k]
            dict[sum] += 1

        return ans

File: leetcode-04/solutions/cpp/0560.cpp
class Solution {
 public:
  int subarraySum(vector<int>& nums, int k) {
    int ans = 0;
    int sum = 0;
    unordered_map<int, int> map;
    map[0] = 1;

    for (int num : nums) {
      sum += num;
      if (map.count(sum - k)) ans += map[sum - k];
      map[sum]++;
    }

    return ans;
  }
};